tests/functional/test_types.py

```diff
@@ -147,3 +149,24 @@ def test_signature_from_rsv_and_vrs(signature):
     from_rsv = signature.from_rsv(rsv)
     from_vrs = signature.from_vrs(vrs)
     assert from_rsv == from_vrs == signature
+
+
+def test_address_type(owner):
+    class MyModel(BaseModel):
+        addr: AddressType
+
+    # Show str works.
+    instance_str = MyModel(addr=owner.address)
+    assert instance_str.addr == owner.address
+
+    # Show hex bytes work.
+    instance_hex_bytes = MyModel(addr=HexBytes(owner.address))
+    assert instance_hex_bytes.addr == owner.address
+
+    # Show raw bytes work.
+    instance_bytes = MyModel(addr=bytes.fromhex(owner.address[2:]))
```

Comment:
@Aviksaikat This is the case you are in.  If you are using raw bytes, make sure it looks the same as doing `bytes.fromhex()` on the hex str _without_ the 0x prefix... or even simpler, you can just use the hex str directly!

thank you

Reply:
In general, we are using Pydantic v2 now, which is less forgiving on various "implicit conversions" that may have been happening before

Reply:
Cool. :+1:

Reply:
Bad things can also happen if we try to handle bytes encoded hex strings like `b"0x123"`, the mixing of the formats leads to ambiguity.